### ISSUE TITLE
PropertyUsedBeforeDeclaration false positive nested usage of constant

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#6125]
## Expected Behavior
No finding, the inner class will be initialized after the outer class. Also since this is a constant, it'll be inlined by the compiler, so the property is "unused" in the bytecode.

## Observed Behavior
```console
PropertyUsedBeforeDeclaration - ['DAGGER_NAME' is before declaration.] at ...
```

## Steps to Reproduce
```kotlin
@Module
object MyModule {
	interface Dependencies<Builder> {
		@BindsInstance
		fun prop(@Named(DAGGER_NAME) namedProp: String): Builder
	}
	internal const val DAGGER_NAME = "dagger-name"
}
```
for reference: `annotation class Named(val value: String)`.

## Context
Strangely
```kotlin
object Outer {
    object Inner {
        val inner = outer
    }
    val outer = "value"
}
```
is fine, so there must be something about the annotation probably.

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): 8.1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows
* Link to your project (if it's a public repository): N/A

cc @t-kameyama as author of the rule https://github.com/detekt/detekt/pull/6062
